### PR TITLE
Website: Intrinsic RevShare Token

### PR DESCRIPTION
**Pull request by socathie**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: ERCS/erc-intrinsic_revshare_token.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
+---
```
Comment (abcoathup) [inline ERCS/erc-intrinsic_revshare_token.md:1]:
```suggestion
---
eip: 7641
```

File: ERCS/erc-intrinsic_revshare_token.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+title: Intrinsic RevShare Token
+description: An ERC-20 extension that integrates a revenue-sharing mechanism, ensuring tokens intrinsically represent a share of a communal revenue pool
+author: Conway (@0x1cc), Cathie So (@socathie), Xiaohang Yu (@xhyumiracle), Suning Yao (@fewwwww), Kartin <kartin@hyperoracle.io>
+discussions-to: https://ethereum-magicians.org/t/erc-intrinsic-revshare-token/18999
```
Comment (abcoathup) [inline ERCS/erc-intrinsic_revshare_token.md:5]:
```suggestion
discussions-to: https://ethereum-magicians.org/t/erc-7641-intrinsic-revshare-token/18999
```

File: ERCS/erc-7641.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+### `claimableRevenue` interface
+
+We specify a `claimableRevenue` interface to calculate the amount of ETH claimable by a token holder at a certain snapshot. We will share the revenue between two consecutive snapshots. Specifically, assuming that the revenue between two snapshots is `R`, we will specify a revenue sharing ratio `p`, ranging from 0%-100%, and we will share the revenue of `pR` to different token holders according to the token ratio. Specifically,  the amount of ETH claimable by a token holder with `amount` tokens at a certain snapshot is `pR * amount / totalAmount` , where `totalAmount`  denotes the total amount of [ERC-7641](./eip-7641.md) token. Noted that the remaining revenue of `(1-p)R` will be retained in the revenue pool, and we can take out this part of revenue through burning.
```
Comment (SamWilsn) [inline ERCS/erc-7641.md:139]: This paragraph sounds like it's creating requirements on implementations? For example, "the amount of ETH claimable [...] is `pR * amount / totalAmount`" seems like an instruction to implementers.

If these are requirements, they'll need to be moved to the Specification section. The _reasoning_ behind why that formula exists should remain in the Rationale section.

The idea is that the Specification section, taken alone, should be enough for someone to write a compliant implementation of this proposal. The other sections (Rationale included) are here for context and history purposes.

File: assets/erc-7641/package-lock.json
Comment (SamWilsn) [inline assets/erc-7641/package-lock.json:1]: Please remove build system specific files from your reference implementation.

File: ERCS/erc-7641.md (hunk 3 above)
Comment (socathie) [reply ERCS/erc-7641.md:139]: It's more of a recommended/reference implementation. How do you recommend us to change the wording here to reflect this?

Comment (socathie) [reply ERCS/erc-7641.md]: We have changed the wording to reflect that it is an example. Please review!

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**